### ISSUE TITLE
[50] Real-IAD: A Real-World Multi-View Dataset for Benchmarking Versatile Industrial Anomaly Detection

### ISSUE THREAD

**Issue by sy00n** [sy00n/DL_paper_review#55]
## Abstract
- Mainstream dataset이 포화되어서 실제 application 시나리오에서의 데이터셋에서 성능 차이가 발생함.
- 또 데이터 스케일 제한으로 인해 overfitting 리스크가 있었음.
- Real-IAD는 150k high-resolution 이미지, 30 종류의 object, 기존보다 커진 규모임. defect 범위도 더 크고 ratio proportion도 더 크기 때문에 기존 데이터셋보다 더 challenging함.
- Real application 시나리오에 가까운 데이터셋을 만들기 위해 multi-view shooting method를 적용했고 sample-level evaluation metrics를 제안함.
- 추가로, general unsupervised anomaly detection setting에서 industrial production의 수율이 60% 보다 높다는 점을 바탕으로 fully unsupervised industrial anomaly detection(FUIAD) 기반의 새로운 셋팅을 제안함.

## Introduction
: 현재 AD dataset의 양은 여전히 작고 추가적인 developement가 필요함. 
- 규모 관련 문제
  - MVTec의 d**efect의 범위는 좁고 단순**함. 최근 방법들은 1/P-AUROC에서 99%를 넘으면서 **새 방법론들의 우열을 가리기가 힘들어짐.**
  - MVTec은 **데이터의 규모도 작음**(defect이 약 60장), test 이미지 수가 적어서 발생하는 **random error**를 무시할 수 없음.
  - MVTec이랑 VisA는 **object 수도 적어서**(각각 15, 12) 모델의 **지표가 지나치게  optimistic하게** 보이게 함. 
- Real application과의 GAP
  - 현재 데이터셋의 설정과 이상적인 application 시나리오 차이에는 갭이 존재함. 비록 IAD가 unsupervised learning이지만, train set에는 **여전히 manual annotation을 필요로 함**. 그래서 Soft-Patch가 제안한 BTAD train set의 noise sample과 같은 **noise sample이 포함될 수 있음.** 
  - 2D IAD 데이터셋은 주로 **single-view 이미지들로 구성**되어 있는데 실제 applications에서 part structure는 복잡하며 single-view로는 모든 defect을 커버할 수 없음.
  - MVTec AD-3D와 같은 일부 데이터셋은 3D 관점에서 multi-view 문제를 해결하려고 시도했지만 3D 센서의 높은 비용으로 인해 practical application을 제한함. 
 

-> 새로운 Real-world Industrial Anomaly Detection 데이터셋인 **Real-IAD**를 제안함.
  - **Object 카테고리 수와 이미지 장 수** 측면에서 기존 데이터셋을 훨씬 능가함.
  - 2D IAD task에서 **multi-view** 문제를 처음으로 고려한 데이터셋임.
  - Practical application을 위한 이상적인 시나리오에서는 이미지 수집 장비가 생산 라인에 배치되고 알고리즘이 자동으로 학습되어 인간의 개입 없이 제품이 결험이 있는지 여부를 판단할 수 있게 되는데 이 때 수율이 60%이상이라는 특성을 바탕으로, **quality inspection 이상 이미지를 0~40% 추가**해서 **fully unsupervised anomaly detection문제**를 처음으로 제안함.
  - **총 30 종류의 제품**을 수집했고 더 미세한 defect 특징을 포착하기 위해 **해상도를 위도적으로 높여서** 고정밀 defect detection을 가능케 함.

## 3. Real-IAD Dataset Description
![image](https://github.com/user-attachments/assets/13bab8fb-f79d-4826-8400-869c45de5360)
### 3.1. Data Collection and Construction Manner
1. Material Preparation: 다양한 재료9metal, plastic, wood, ceramics, and mixed materials)를 포함한 30개의 obejct를 수집. 여러 종류의 defect(missing parts, dirt, deformation, pits, damage, holes, cracks, scratches)을 수동으로 생성.
2. Prototype Construction: 다섯 대의 카메라를 사용하여 다양한 각도에서 objects 촬영
3. Data Collection, Annotation, and Cleaning: 모든 normal 및 anomaly images 수동 확인 및 LabelMe를 사용한 pixel-level mask 레이블링 수행. cascade RCNN based on HRNet-32w backbone으로 학습을 시켜서 모델 예측 결과와 manual annotation 결과가 다른 경우 모델의 예측 AP 결과가 달라지지 않을 때까지, 특정 개수 이하로 떨어질 때까지 검토하고 다시 annotation 함. 

### 3.2. Comparison with Popular 2D Datasets
- Mainstream Datasets Comparison: Real-IAD는 카테고리 수에서 최소 2배 증가, 이미지 수는 10K에서 150K로 증가. 최대 2,000∼5,000 해상도의 이미지 제공, multi-view 이미지 포함.
- Data Statistics: 정상 및 이상 데이터 양에서 큰 개선, defect 영역 비율 및 범위도 더 큼. Normal과 abnormal data의 비율은 크게 차이나지 않음.
- Advantage Analyses: 다양성, 대규모, 높은 난이도 측면에서 장점.

### 3.3. Real-IAD Visualization
Real-IAD는 다양한 재료와 결함 유형의 데이터를 포함하며, 결함 영역 비율은 0.1에서 0.5, 비율은 0.1에서 10.0까지 다양. 이는 Real-IAD가 매우 challenging함을 나타냄.
![image](https://github.com/user-attachments/assets/11419523-1411-43f3-a854-1c01ad335ce7)

![image](https://github.com/user-attachments/assets/74f3fd25-6455-4abb-a91e-e9f66c77d75e)
![image](https://github.com/user-attachments/assets/7741c8e2-5a74-4cfb-bc4a-1a06d787f40d)

### 4.1. Evaluation Settings
- Unsupervised IAD (UIAD): 정상 샘플에서만 학습. 기존 anomaly datasets는 일반적으로 학습 세트에는 정상 샘플만 포함하고, 테스트셋은 정상 및 anomaly 샘플 모두를 포함함. 
- Fully Unsupervised IAD (FUIAD): 실제 상황에서 불가피하지만 잘 논의되지 않음. 기존 datasets에서 anomaly 샘플의 제한으로 FUIAD 설정을 구성하기 어려움. Real-IAD는 다양하고 큰 anomaly 샘플로 FUIAD 설정을 구축하였는데테스트셋에는 100개의 정상 + anomaly 샘플(총 500장)을 고정하고, 남은 샘플로 noisy training set을 구성한다. 다양한 noise ratios로 새로운 fully unsupervised benchmarks를 생성.

### 4.2. Evaluation Metric
- Multi-view evaluiation: Real-IAD는 최초의 multi-view anomaly detection dataset으로, 여러 view 에서의 결과를 통합해 샘플 수준의 성능 평가. 이는 산업 생산 라인의 평가와 더 유사함.

## 5. Comparisons with IAD Benchmarks
![image](https://github.com/user-attachments/assets/8638ee8b-bf3b-46e5-b81e-eae922bae859)

### 5.1. Results on Unsupervised IAD
- MVTec(97.9% I-AUROC)에서 Real-IAD(single-view 85% I-AUROC, multi-view)로 성능이 크게 감소, 이는 Real-IAD가 더 challenging함을 의미
- Real-IAD의 픽셀 수준 P-PRO는 VisA와 비슷하지만 MVTec AD보다는 낮음, 이는 pixel-level 이상 탐지에서도 challenging 함을 의미

### 5.2. IAD Benchmarks Meet FUIAD
noise ratios: 학습 데이터셋에서 정상 샘플에 비정상 샘플이 포함되 비율을 의미함. 만약 0.1이면 학습 데이터셋의 10%가 anomaly라는 의미임. 
![image](https://github.com/user-attachments/assets/e25682b7-e4a8-47c8-bfc2-30c5b68a4a7e)
![image](https://github.com/user-attachments/assets/9c2cd7e7-23de-4814-b5d9-e50a0c553744)

- UIAD 설정(noise ratio 0)에서는 대부분의 이상 탐지 방법들이 유사한 성능을 보임.
- pixel-level anomaly segmentation에서는 PatchCore, SoftPatch, RD가 multi-level feature을 사용해 유리함.
- FUIAD 설정에서는 대부분의 방법이 모든 지표에서 성능 저하, 특히 PaDim, SimpleNet, DeSTSeg, RD에서 많이 떨어짐.
- PatchCore는 patch-level Memory-Bank로 인해 robust함. Memory-Bank에 정상 및 비정상 특징을 동시에 저장하여 inference시에 저장된 이상이랑 다른 경우의 이상이 발생한 경우 detection이 가 능함. 
  - 그리고 noise ratio가 0.4처럼 좀 높더라도 비정상 픽셀의 비율은 매우 낮기 때문에 memory bank에서 정상 패치 품질을 어느정도 보장할 수 있음.
- SoftPatch는 noise feature를 먼저 필터링한 후 Memory-Bank를 구축하기 때문에 성능이 가장 높음.
그러나 비지도 PatchCore에 비해 성능 향상 폭은 limited 이는 FUIAD가 더 심도 있는 연구가 필요함을 의미. (여기서 에시로는 앙상블 방법이나 LLM을 결합해서 노이즈 데이터의 prior distribution을 제공하는 등의 방법이 필요하다고 함)

## 결론
- 기존의 문제
  - unsupervised industrial anomaly detection의 성능은 거의 포화 상태에 도달했지만, actual industrial
  inspection에 적용하기는 어려움.
  - 대부분의 알고리즘은 clean normal training sample 에 의존하지만, 실제 산업 생산 라인의 데이터에는 일정량의 노이즈가 포함됨.
  - FUIAD는 실제 어플리케이션에 더 적합하지만, 기존 데이터셋은 샘플이 제한적임.
  
- 제안하는 Real-IAD
  -150k 고해상도 이미지를 포함한 large-scale, real-world, multi-view anomaly detection datasetReal-IAD 제안.
  - 30개의 metal, plastic, wood, ceramics, and mixed materials, 각 객체는 8개의 일반적인 defect을 포함한 5개의 다른 view로 구성.
  - pixel, image, sample level에서 high-quality annotation 제공.
  - unsupervised/ fully unsupervised 세팅을 통해 최신 이상 탐지 방법론들을 평가.
